_studio/shared/src/libmfx_core.cpp

```diff
@@ -874,14 +897,16 @@ mfxBaseWideFrameAllocator* CommonCORE::GetAllocatorByReq(mfxU16 type) const
 mfxStatus CommonCORE::SetFrameAllocator(mfxFrameAllocator *allocator)
 {
     UMC::AutomaticUMCMutex guard(m_guard);
-    if (!allocator)
-        return MFX_ERR_NONE;
 
     if (!m_bSetExtFrameAlloc)
     {
-        m_FrameAllocator.frameAllocator = *allocator;
         m_bSetExtFrameAlloc = true;
-        m_session->m_coreInt.FrameAllocator = *allocator;
+        if (allocator) {
```

Comment:
most of `if (m_bDelayedFrameAllocation)` in this PR  can be removed, if we implement and set an internal DealyedFrameAllocator here.

Reply:
AFAIU you use SetFrameAllocator(allocator = NULL) as a trigger to enable this mode. I'd like to have something explicit  - separate mode DelayedFrameAllocation as we talked before.

Reply:
Hi @varistar , 
the allocator or getHDL is a session/core level function. It will shared by decoder/encoder/vpp/. We can't use a decoder param to signal it. If we do not use SetFrameAllocator(allocator = NULL), we need add other function like  MFXVideoCORE_SetFrameAllocatMode(mfxSession sessio, MODE mode).

Reply:
Hi @varistar,
Are you prefer add new MFXVideoCORE function  or you have other insights on this?
thanks